keras_cv/layers/preprocessing3d/group_points_by_bounding_boxes.py

```diff
@@ -0,0 +1,199 @@
+            bounding_boxes_mask = ~tf.math.equal(bounding_boxes[0, :, 7], 0.0)
+            object_bounding_boxes = tf.boolean_mask(
+                bounding_boxes, bounding_boxes_mask, axis=1
+            )
+
+        points_in_bounding_boxes = is_within_box3d(
+            point_clouds, object_bounding_boxes[:, :, :7]
+        )
+        # Filter bounding boxes using the current frame.
+        min_points_filter = (
+            tf.reduce_sum(tf.cast(points_in_bounding_boxes[0], dtype=tf.int32), axis=0)
+            >= self._min_points_per_bounding_boxes
+        )
+
+        object_bounding_boxes = tf.boolean_mask(
+            object_bounding_boxes, min_points_filter, axis=1
+        )
+
+        points_in_bounding_boxes = tf.boolean_mask(
+            points_in_bounding_boxes, min_points_filter, axis=2
+        )
+        # [num of frames, num of boxes, num of points].
+        points_in_bounding_boxes = tf.transpose(points_in_bounding_boxes, [0, 2, 1])
+        points_in_bounding_boxes = tf.cast(points_in_bounding_boxes, tf.int32)
+        sort_valid_index = tf.argsort(
+            points_in_bounding_boxes, axis=-1, direction="DESCENDING"
+        )
+        sort_valid_mask = tf.gather(
+            points_in_bounding_boxes, sort_valid_index, axis=2, batch_dims=2
+        )[:, :, : self._max_points_per_bounding_boxes]
+        # [num of frames, num of boxes, self._max_points_per_bounding_boxes, num of point features].
+        object_point_clouds = point_clouds[:, tf.newaxis, :, :]
+        num_valid_bounding_boxes = tf.shape(object_bounding_boxes)[1]
+        object_point_clouds = tf.tile(
+            object_point_clouds, [1, num_valid_bounding_boxes, 1, 1]
+        )
+        object_point_clouds = tf.gather(
+            object_point_clouds, sort_valid_index, axis=2, batch_dims=2
+        )[:, :, : self._max_points_per_bounding_boxes, :]
+
+        object_point_clouds = tf.where(
+            sort_valid_mask[:, :, :, tf.newaxis] > 0, object_point_clouds, 0.0
+        )
+
+        return (
+            object_point_clouds,
+            object_bounding_boxes,
+        )
+
+    def _augment(self, inputs):
+        point_clouds = inputs[POINT_CLOUDS]
+        bounding_boxes = inputs[BOUNDING_BOXES]
+
+        transformation = self.get_random_transformation(
+            point_clouds=point_clouds,
+            bounding_boxes=bounding_boxes,
+        )
+        (
+            object_point_clouds,
+            object_bounding_boxes,
+        ) = self.augment_point_clouds_bounding_boxes(
+            point_clouds,
+            bounding_boxes=bounding_boxes,
+            transformation=transformation,
+        )
+
+        result = {
+            OBJECT_POINT_CLOUDS: object_point_clouds,
+            OBJECT_BOUNDING_BOXES: object_bounding_boxes,
+        }
+
+        # preserve any additional inputs unmodified by this layer.
+        for key in inputs.keys() - result.keys():
+            result[key] = inputs[key]
+        return result
+
```

Comment:
nit: maybe you can start with results = inputs, and then results.update({OBJECT_POINT_CLOUD, OBJECT_BOUNDING_BOXES})

Reply:
Done.